src/kit/LogViewer/LogViewer.tsx

```diff
@@ -350,30 +267,44 @@ const LogViewer: React.FC<Props> = ({
   }, [logs]);
 
   const handleFullScreen = useCallback(() => {
-    if (baseRef.current && screenfull.isEnabled) screenfull.toggle();
+    if (logsRef.current && screenfull.isEnabled) screenfull.toggle();
   }, []);
 
   const handleDownload = useCallback(() => {
     onDownload?.();
   }, [onDownload]);
 
-  // Fetch initial logs on a mount or when the mode changes.
+  // Re-fetch logs when fetch callback changes.
+  useEffect(() => {
+    local.current = clone(defaultLocal);
+
+    setLogs([]);
+    setIsTailing(true);
+    setFetchDirection(FetchDirection.Older);
+    setFirstItemIndex(START_INDEX);
+  }, [onFetch]);
+
+  // Add initial logs if applicable.
+  useEffect(() => {
+    if (!initialLogs) return;
+    addLogs(processLogs(initialLogs.map((log) => decoder(log))));
+  }, [addLogs, decoder, initialLogs, processLogs]);
+
+  // Initial fetch on mount or when fetch direction changes.
   useEffect(() => {
     fetchLogs({ canceler, fetchDirection }, FetchType.Initial).then((logs) => {
-      addLogs(logs, true);
-
-      if (fetchDirection === FetchDirection.Older) {
-        // Slight delay on scrolling to the end for the log viewer to render and resolve everything.
-        setTimeout(() => {
-          listRef.current?.scrollToItem(Number.MAX_SAFE_INTEGER, 'end');
-          local.current.isScrollReady = true;
-        }, 100);
-      } else {
-        listRef.current?.scrollToItem(0, 'start');
+      addLogs(logs, false);
+
+      // Slight delay on scrolling to the end for the log viewer to render and resolve everything.
+      setTimeout(() => {
+        virtuosoRef.current?.scrollToIndex({
+          index: fetchDirection === FetchDirection.Older ? 'LAST' : 0,
+        });
         local.current.isScrollReady = true;
-      }
+      }, 200);
     });
-  }, [addLogs, canceler, fetchDirection, fetchLogs]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why are we skipping dependencies here?

Reply:
I wanted to limit the initial log fetch to only happen on mount and when the direction changed and was concerned that the useEffect would fire too often with the full dep array.

Reply:
I think doing that broke searching through, so I'll reverse it.